### PR TITLE
Update dev env instructions

### LINKED ISSUE #2793: explore `pixi` for sourmash
see: https://github.com/sourmash-bio/sourmash/issues/2707#issuecomment-1686499578

### PR DESCRIPTION
**Pull request by luizirber**
Close #2793 

- Add pixi, first mentioned in https://github.com/sourmash-bio/sourmash/issues/2707#issuecomment-1686499578
- Use tabs in docs to help to point out only one option is needed
- Default recommendation is conda, add `dev.yml` with dev dependencies

### CODE REVIEW

**Comment by ctb** [comment]
I mean I've loved pixi, the little bit I've used it; but do we really want to switch dev instructions over to it?

**Comment by ctb** [comment]
(I guess I'm not immediately seeing a big reason. But I am happy to be convinced :)

**Comment by luizirber** [comment]
> I mean I've loved pixi, the little bit I've used it; but do we really want to switch dev instructions over to it?

We have the drift on install instructions (as pointed in #2707) because what we document is not quite we run in CI. 

I was going for setting up pixi and using the `pixi.lock` file to also install a conda environment if pixi is not available, but `conda-lock` and `pixi` diverged on the lockfile format. Maybe they will [converge](https://github.com/mamba-org/rattler/issues/465#issuecomment-1876908565) back, but for now they are incompatible.

It is still fine to have conda instructions and say "look at the `[dependencies]` section in pixi.toml and run `conda create` with that", but we start getting into a lot of duplicate dependency definitions... We could have an `environment.yml` and keep it in sync with pixi.toml too, if that's easier.

> (I guess I'm not immediately seeing a big reason. But I am happy to be convinced :)

I like pixi because it is so much easier to install than conda, and doesn't mess with shell env vars. And if you're in a system that has conda installed but doesn't allow extra software installation, you can still [`conda install -c conda-forge pixi`](https://github.com/conda-forge/pixi-feedstock) and have it available.

Drawback is that conda is much more stable, pixi has been moving pretty quickly and might be hard to keep up (but we do have CI setup in this PR).

It boils down to which is the easiest one to teach and get started, pixi seems easier because it's one file to download and `pixi shell` to get started, but conda can be more general-purpose and transfer better to other situations?

All that said, I don't use conda or pixi for sourmash development, so better to someone that does use to chime in 🙃

File: doc/developer.md [hunk 1]
```diff
@@ -12,33 +12,28 @@ git clone https://github.com/sourmash-bio/sourmash.git
 ```
 sourmash runs under Python 3.10 and later.
 
-We recommend using `conda` or `Nix` for setting up an environment for developing
+We recommend using `pixi` or `Nix` for setting up an environment for developing
 new features, running tests and code quality checks.
 Here are some suggestions on how to set them up (note: you only need one =])
 
-### Using mamba (conda alternative)
+::::{tab-set}
```
Comment (luizirber) [inline doc/developer.md:23]: How the tabs look like: https://sourmash--2990.org.readthedocs.build/en/2990/developer.html
image

Comment (ctb) [reply doc/developer.md:23]: I love it!

**Review by ctb — APPROVED** [review]

**Pull request merged**